test/IdleMStableStrategy.js

```diff
@@ -0,0 +1,168 @@
+    expect(await imUSD.name()).to.eq("Interest bearing mUSD");
+    expect(await mUSD.name()).to.eq("mStable USD");
+    expect(await meta.name()).to.eq("Meta");
+    expect(await vault.boostDirector()).to.eq("0xBa05FD2f20AE15B0D3f20DDc6870FeCa6ACd3592");
+  });
+
+  it("Deposit AMOUNT in Idle Tranche", async () => {
+    let totalSharesBefore = await IdleMStableStrategy.totalSupply();
+
+    // approve tokens to idle-mstable-strategy
+    await mUSD.connect(user).approve(IdleMStableStrategy.address, AMOUNT_TO_TRANSFER);
+    await IdleMStableStrategy.connect(user).deposit(AMOUNT_TO_TRANSFER);
+
+    let totalSharesAfters = await IdleMStableStrategy.totalSupply();
+
+    // strategy should not have any mUSD left
+    expect(await mUSD.balanceOf(IdleMStableStrategy.address)).to.eq(0);
+
+    expect(totalSharesAfters.sub(totalSharesBefore)).gt(0);
+
+    // check states in vault
+    expect(await vault.rawBalanceOf(IdleMStableStrategy.address)).gt(0);
+  });
+
+  it("Redeem Tokens: redeem", async () => {
+    let strategySharesBefore = await IdleMStableStrategy.totalSupply();
+
+    await mUSD.connect(user).approve(IdleMStableStrategy.address, AMOUNT_TO_TRANSFER);
+    await IdleMStableStrategy.connect(user).deposit(AMOUNT_TO_TRANSFER);
+
+    let strategySharesAfter = await IdleMStableStrategy.totalSupply();
+
+    let sharesReceived = strategySharesAfter.sub(strategySharesBefore);
+    expect(sharesReceived).gt(0);
+
+    let redeemAmount = sharesReceived.div(10); // claim back a fraction of shares received
+
+    await IdleMStableStrategy.connect(user).redeem(redeemAmount);
+  });
+
+  it("Redeem Rewards", async () => {
+    let strategySharesBefore = await IdleMStableStrategy.totalSupply();
+
+    await mUSD.connect(user).approve(IdleMStableStrategy.address, AMOUNT_TO_TRANSFER);
+    await IdleMStableStrategy.connect(user).deposit(AMOUNT_TO_TRANSFER);
+
+    let strategySharesAfter = await IdleMStableStrategy.totalSupply();
+
+    let sharesReceived = strategySharesAfter.sub(strategySharesBefore);
+    expect(sharesReceived).gt(0);
+
+    await IdleMStableStrategy.connect(user)["redeemRewards()"]();
+  });
+
+  it("APR", async () => {
+    const musdSwapingContract = await ethers.getContractAt(masset.abi, mUSD.address);
+
+    await mUSD.connect(user).approve(IdleMStableStrategy.address, AMOUNT_TO_TRANSFER);
+    await IdleMStableStrategy.connect(user).deposit(AMOUNT_TO_TRANSFER);
+
+    await mUSD.connect(musd_signer).transfer(user.address, AMOUNT_TO_TRANSFER);
+    await DAI.connect(dai_signer).transfer(user.address, AMOUNT_TO_TRANSFER);
+
+    await DAI.connect(user).approve(musdSwapingContract.address, AMOUNT_TO_TRANSFER);
+
+    await musdSwapingContract.connect(user).swap(DAIAddress, USDCAddress, AMOUNT_TO_TRANSFER.div(2), 0, user.address);
+
+    await savingsManager.connect(user).collectAndStreamInterest(mUSD.address);
+    
+    console.log("APR", (await IdleMStableStrategy.getApr()).toString());
```

Comment:
same as above you should add an assertion

Reply:
Added the assertion a simple assertion, as it is very difficult to predict the APR.